File: APRIL Lunchtime/bench_press.py
t = int(input())
for _ in range(t):
    n, w, wr = map(int, input().split())
    li = list(map(int, input().split()))
    unique_weights = dict()
    set_of_weights = set(li)
    for weight in li:
        if unique_weights.get(weight):
            unique_weights[weight] += 1
        else:
            unique_weights[weight] = 1 
    sum_re = w-wr
    ans= 0
    for temp in unique_weights.keys():
        if unique_weights[temp] > 1:
            if unique_weights[temp]%2==0:
                ans += temp*unique_weights[temp]
            else:
                ans += temp*(unique_weights[temp]-1)
    if ans >= sum_re:
        print("YES")
    else:
        print("NO")

File: APRIL Lunchtime/benchpress.cpp
#include<iostream>
#include<vector>
#include<set>
#include<map>
#define vi vector<int>
#define loop(j, n) for(int i=j; i<n; i++)
#define int long long
using namespace std;
int32_t main()
{
    int t;
    cin >> t;
    while(t--)
    {
        int n, w, wr;
        cin >> n >> w >> wr;
        vi weights(0);
        loop(0, n)
        {
            int temp;
            cin >> temp;
            weights.push_back(temp);
        }
        map<int, int> frequency;
        int ans=0;
        loop(0, n)
        {
            // if(frequency.find(weights[i])!=frequency.end()) {frequency[weights[i]]++;}
            // else frequency[weights[i]]=1;
            frequency[weights[i]]++;
        }
        int sum_required = w-wr;
        auto it = frequency.begin();
        while(it != frequency.end())
        {
            if(it->second %2 == 0) {ans+= it->second * it->first;}
            else {ans+= (it->first)*((it->second)- 1);}
            // cout << "key " << it->first << " " << "value " << it->second << endl;
            it++;
        }
        // cout << "answer " << ans << endl;
        ans>=sum_required ? cout << "YES" << endl : cout << "NO" << endl;
        // cout << "----------------------------------" << endl;
        
    }
    return 0;
}

